assets/components/src/text-control/index.js

```diff
@@ -19,13 +20,31 @@ import './style.scss';
 import classNames from 'classnames';
 
 class TextControl extends Component {
+	constructor( props ) {
+		super( props );
+		this.wrapperRef = createRef();
+	}
+	componentDidMount() {
+		if ( this.wrapperRef.current ) {
+			const labelEl = this.wrapperRef.current.querySelector( 'label' );
+			if ( labelEl ) {
+				labelEl.setAttribute( 'data-required-text', __( '(required)', 'newspack' ) );
+			}
+		}
+	}
 	/**
 	 * Render.
 	 */
 	render() {
-		const { className, ...otherProps } = this.props;
+		const { className, required, ...otherProps } = this.props;
 		const classes = classNames( 'newspack-text-control', className );
-		return <BaseComponent className={ classes } { ...otherProps } />;
+		return required ? (
+			<div ref={ this.wrapperRef } className={ classes }>
+				<BaseComponent required={ required } { ...otherProps } />
```

Comment:
The current placement of `className` on require text controls causes `.newspack-text-control` not to be added to `div.components-base-control`, which makes the labels the wrong font size -- they're `13px` rather `14px`:

![image](https://user-images.githubusercontent.com/177561/89585938-9b416380-d7f3-11ea-99f5-fa5ff2e36e1b.png)

Moving the `className` to the `BaseComponent` seems to fix, but I'm not sure if this is a bad idea for a different reason:

```suggestion
			<div ref={ this.wrapperRef }>
				<BaseComponent className={ classes } required={ required } { ...otherProps } />
```

Reply:
Done in 5ae9711750d7e43a80fd15b32bcfed700c7e849b

Reply:
Looks good, thanks!